spinner/spinner.go

```diff
@@ -23,12 +22,14 @@ import (
 // ⣾  Loading...
 type Spinner struct {
 	spinner    spinner.Model
-	action     func()
+	action     func() error
```

Comment:
I would expect to receive either the context or spinner as argument.


```suggestion
	action     func(ctx context.Context) error
```


```suggestion
	action     func(s *Spinner) error
```

It would help. I only expect to get the context. But maybe the spinner could be useful